### PR TITLE
54 api unit tests auth

### LINKED ISSUE #54: API Unit Tests: Auth
- [ ] Test all validation
- [ ] Test all edge cases (trying to create a user with a taken username, login with incorrect password, etc...)

### PR DESCRIPTION
**Pull request by rdhillon1016**

### CODE REVIEW

File: api/bin/www [hunk 1]
```diff
@@ -8,6 +8,13 @@ const debug = require('debug')('api:server');
 const http = require('http');
 const app = require('../app');
 
+/**
```
Comment (rdhillon1016) [inline api/bin/www:11]: Idk if this is the best way to do this (separating out the db-setup and the "app" object for testing purposes)

**Comment by rdhillon1016** [comment]
We'll need to change the error checks to be more specific in the tests eventually.

Right now it's just checking for 404 codes, but those could be failing for any reason.

**Review by brandhawa99 — COMMENTED** [review]
Looks good but I feel like i added too many things to the user model. Are you okay with removing first name last name and birthday or should we just keep them

**Comment by rdhillon1016** [comment]
> Looks good but I feel like i added too many things to the user model. Are you okay with removing first name last name and birthday or should we just keep them

Yea I'll take them out

**Comment by brandhawa99** [comment]
> > Looks good but I feel like i added too many things to the user model. Are you okay with removing first name last name and birthday or should we just keep them
> 
> Yea I'll take them out

kk Ill write a script after thtat  to fill the database with like 20 or so users with posts
